File: packages/appwsgi/415.tar.bz2/trunk/www/smtp/smtp.js
// Copyright(c) gert.cuykens@gmail.com
smtp=
{
 'send':function()
 {
  http.fml=function(xml)
  {
   //alert(xml.getElementsByTagName('error')[0].childNodes[0].nodeValue)
  }
  http.xml='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <user>'+document.getElementById('user').value+'</user>\n'
  http.xml+=' <password>'+document.getElementById('password').value+'</password>\n'
  http.xml+=' <to>'+document.getElementById('to').value+'</to>\n'
  http.xml+=' <subject>'+document.getElementById('subject').value+'</subject>\n'
  http.xml+=' <message>'+document.getElementById('message').value+'</message>\n'
  http.xml+='</root>'
  http.send()
 }
}

File: packages/appwsgi/appwsgi%201057.tar.bz2/appwsgi/www/smtp/smtp.py
# Copyright(c) gert.cuykens@gmail.com
from frame import Frame
from smtplib import SMTP

def application(environ, response):
    xml = environ['wsgi.input'].read()
    v   = Frame.read(xml)
    xml = '<?xml version="1.0" encoding="UTF-8"?>\n'
    xml+= '<root>\n'
    xml+= mail(v['user'],v['password'],v['to'],v['subject'],v['message'])
    xml+= '</root>'
    response('200 OK', [('Content-type', 'text/xml')])
    return [xml]

def mail(f,p,t,s,m):
    m = "From: %s\r\nTo: %s\r\nSubject: %s\r\n\r\n%s" % (f, t, s, m)
    try :
        server = SMTP('smtp.gmail.com', 587)
        server.set_debuglevel(1)
        server.ehlo()
        server.starttls()
        server.ehlo()
        server.login(f,p)
        server.sendmail(f,t,m)
        #server.rset()
        #server.quit()
    except :
        return ' <error>mail not send</error>\n'
    return ' <error>send succesfull</error>\n'

